consensus/types/src/beacon_block_body.rs

```diff
@@ -251,6 +251,97 @@ impl<E: EthSpec> From<BeaconBlockBodyMerge<E, FullPayload<E>>>
     }
 }
 
+// We can clone a full block into a blinded block, without cloning the payload.
+impl<E: EthSpec> BeaconBlockBodyBase<E, FullPayload<E>> {
+    pub fn clone_as_blinded(&self) -> BeaconBlockBodyBase<E, BlindedPayload<E>> {
+        let BeaconBlockBodyBase {
+            randao_reveal,
+            eth1_data,
+            graffiti,
+            proposer_slashings,
+            attester_slashings,
+            attestations,
+            deposits,
+            voluntary_exits,
+            _phantom,
+        } = self;
+
+        BeaconBlockBodyBase {
+            randao_reveal: randao_reveal.clone(),
+            eth1_data: eth1_data.clone(),
+            graffiti: *graffiti,
+            proposer_slashings: proposer_slashings.clone(),
+            attester_slashings: attester_slashings.clone(),
+            attestations: attestations.clone(),
+            deposits: deposits.clone(),
+            voluntary_exits: voluntary_exits.clone(),
+            _phantom: PhantomData,
+        }
```

Comment:
We can save a little bit of code duplication here by leaning on the implementation that converts a full `BeaconBlockBodyBase` to a blinded `BeaconBlockBodyBase`. I think this is equivalent in terms of efficiency, because we need to clone all the fields of the `BeaconBlockBodyBase`, and the `into` will just move the cloned fields into the new container (which should probably just be elided by the compiler altogether).

```suggestion
        self.clone().into()
```